.travis.yml

```diff
@@ -37,20 +37,26 @@ matrix:
       language: minimal
       install:
         - docker-compose build
+        - pip install codecov --user
       script:
-        - docker-compose run backend ./runtests.sh
+        - docker-compose run backend bash -c "pytest --cov=./ --cov-report=xml" ./runtests.sh
+      after_success :
+        - codecov -f coverage.xml -t 995256be-3a1f-4d10-b181-e24f88603b6b
 
     - env: python-tests-main
       language: minimal
       install:
         - docker-compose build
+        - pip install codecov --user
       script:
         # Several security features in settings.py (eg setting HSTS headers) are conditional on
         # 'https://' being in the site URL. In addition, we override the test environment's debug
         # value so the tests pass. The real environment variable will be checked during deployment.
         - docker-compose run -e SITE_URL=https://treeherder.dev -e TREEHERDER_DEBUG=False backend python -bb ./manage.py check --deploy --fail-level WARNING
         # Using `-bb` mode to surface BytesWarnings: https://docs.python.org/3.7/using/cmdline.html#cmdoption-b
-        - docker-compose run backend python -bb -m pytest tests/ --runslow --ignore=tests/selenium/
+        - sudo docker-compose run backend bash -c "pytest --cov=./ --cov-report=xml" python -bb -m pytests tests/ --runslow --ignore=tests/selenium
+      after_success :
+        - codecov -f coverage.xml -t 995256be-3a1f-4d10-b181-e24f88603b6b
```

Comment:
What do I do to generate a key for the Treeherder codecov account?

Reply:
You can click on the settings tab (right top corner) on the Treeherder codecov account page. You will find the key over there.

Reply:
Thank you!

I noticed this information:
> Note: Token not required for public repositories uploading from Travis, CircleCI or AppVeyor.

That probably means that you don't need to specify it.